index.js

```diff
@@ -3005,25 +3005,41 @@ export async function createRustBom(path, options) {
       }
     }
 
-    // Dependencies from multiple Cargo.lock files may have an overlap in
-    // dependencies. So the dependency tree may have duplicates which needs
-    // to be removed.
-    const dependencyInList = (list, dependency) =>
-      list.some((pkg) => pkg.ref === dependency.ref);
-    const uniqueDependencyList = dependencyTree.reduce(
-      (accumulator, currentDependency) => {
-        if (!dependencyInList(accumulator, currentDependency)) {
-          accumulator.push(currentDependency);
-        }
-        return accumulator;
-      },
-      []
+    // This is almost a reimplementation of the function mergeDependencies.
+    // It's reimplemented here for the following reasons:
+    //  * mergeDependencies will not merge dependencies without a
+    //    BOM-reference of the component being described. Often
+    //    called "parent component", found in the BOM at metadata.component.
+    //  * This implementation does not look for a suitable BOM-reference for
+    //    the component being described. A component's PURL is often assigned
+    //    as BOM-reference and the parent component's PURL does not appear
+    //    inside Cargo.lock-files, and no package name is looked for in
+    //    Cargo.toml-files.
+    const packageDependencyMap = {};
+    dependencyTree.forEach((pkg) => {
+      if (!packageDependencyMap[pkg.ref]) {
+        packageDependencyMap[pkg.ref] = new Set();
+      }
+      pkg.dependsOn.forEach((dependency) => {
+        packageDependencyMap[pkg.ref].add(dependency);
+      });
+    });
+    const uniqueDependencyTree = Object.entries(packageDependencyMap).map(
+      (entry) => {
+        const [ref, dependencies] = entry;
+        const dependsOn = Array.from(dependencies).sort();
+        dependsOn.sort();
+        return {
+          ref,
+          dependsOn
+        };
+      }
     );
 
     return buildBomNSData(options, pkgList, "cargo", {
       src: path,
       filename: cargoLockFiles.join(", "),
-      dependencies: uniqueDependencyList
+      dependencies: uniqueDependencyTree
```

Comment:
Yes. Actually, it *would* be as simple as calling `mergeDependencies` with a nonsensical but truthy `bom-ref` value. But I think that seems like an unstable way to go.

I think solving #939 is the way to go in order to use `mergeDependencies`. I can work it in to this PR if you'd like. But I am a bit unsure of how to correctly interpret Rust projects' structure into the CycloneDX format or a BOM in general. I would appreciate multiple views on how that should work. My current understanding is this:

* It seems easy, but not necessarily correct, to assume the `[package]` section of the topmost `Cargo.toml` to reflect the entire subject of the BOM. It may be problematic because there may be several products inside a single project.
* If I understand [Cargo's documentation](https://doc.rust-lang.org/cargo/guide/project-layout.html) correctly, `Cargo.toml` and `Cargo.lock` *should* be in the root of the project, and it *should* contain a `[package]` section representing the project (setting aside [workspaces](https://doc.rust-lang.org/cargo/reference/workspaces.html) for a short while). That seems to me as a simplification Cargo can do, but something a BOM may not be able to do. In the case of [Cargo itself](https://github.com/rust-lang/cargo/blob/master/Cargo.toml), there should be a different BOM for Cargo, the binary application, and `cargo`, the library.

Specifically for dependencies, this does not seem to be an issue. [Cargo's documentation for *Library* (singular)](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#library) and [discussions](https://internals.rust-lang.org/t/how-about-changing-lib-to-lib-to-allow-multiple-library-in-a-crate/2022/2) for allowing multiple libraries in a single project suggests Cargo's simplification works well for dependencies. But it is still an issue for multiple binaries leading to multiple BOM subjects in a single project.

Reply:
@Lucasljungberg, to prevent this PR from becoming a long running can we create a default parent component based on the directory name alone? And then look into constructing the proper metadata.component.components (parent component that includes the various workspaces) as a separate PR?

Reply:
Okay, I'll fix it some time today!

Reply:
We can include this as a feature of 10.3.0. Thank you so much!